### PR TITLE
Allow resource to be directly nested under the api element

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

**Comment by pksunkara** [comment]
One of you need to check the 2nd commit closely. It had almost rewritten test fixtures. I could have added a resource group.

**Comment by pksunkara** [comment]
I added resource group to most of the fixtures because otherwise the whole JSON output of a fixture will change and it lead to a lot of noise in the commit.

File: src/utils/so/JsonIo.cc [hunk 1]
```diff
@@ -85,7 +85,7 @@ namespace
                     break;
                 default: {
                     // unsigned int representation
-                    std::uint8_t c = byte < 0x80 ? byte : byte + 0x80;
+                    std::uint8_t c = static_cast<std::uint8_t>(byte);
```
Comment (klokane) [inline src/utils/so/JsonIo.cc]: Why this?

**Comment by klokane** [comment]
All other seems ok

File: src/utils/so/JsonIo.cc (hunk 1 above)
Comment (pksunkara) [reply src/utils/so/JsonIo.cc]: It's not compiling on mac without the changes here.

**Pull request merged**